packages/block-editor/src/components/global-styles/use-global-styles-output.js

```diff
@@ -395,21 +381,40 @@ export function getStylesDeclarations(
 	);
 
 	/*
-	 * Set background defaults.
-	 * Applies to all background styles except the top-level site background.
+	 * Preprocess background image values.
+	 *
+	 * Note: As we absorb more and more styles into the engine, we could simplify this function.
+	 * A refactor is for the style engine to handle ref resolution (and possibly defaults)
+	 * via a public util used internally and externally. Theme.json tree and defaults could be passed
+	 * as options.
 	 */
-	if ( ! isRoot && !! blockStyles.background ) {
-		blockStyles = {
-			...blockStyles,
-			background: {
-				...blockStyles.background,
-				...setBackgroundStyleDefaults( blockStyles.background ),
-			},
-		};
+	if ( !! blockStyles.background ) {
+		/*
+		 * Resolve dynamic values before they are compiled by the style engine,
+		 * which doesn't (yet) resolve dynamic values.
+		 */
+		if ( blockStyles.background?.backgroundImage ) {
+			blockStyles.background.backgroundImage = getResolvedValue(
+				blockStyles.background.backgroundImage,
+				tree
+			);
+		}
+
+		/*
+		 * Set default values for block background styles for uploaded images.
+		 * Top-level styles are an exception as they are applied to the body.
+		 */
+		if ( ! isRoot && !! blockStyles.background?.backgroundImage?.id ) {
```

Comment:
Not necessarily something to worry about just now, but what if we allowed external urls at some point via the UI? I suppose in that case, we could set the defaults when the user adds the external url, rather than worrying about it at this part of the process. So I think that means this code is probably fine!

Reply:
That's a good 	question. Yeah, I like your idea of setting some defaults here. I'm sure we'll find a way to identify what's been added by the user in the editor. 

I think 🙃

Reply:
Yeah, totally. Again, not a blocker for now!